File: src/ServerlessApi/BackendFunctions/Context/TelemetryRecord.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ServerlessApi.Context
{
    public class TelemetryRecord
    {
        public Guid Id { get; set; }
        public Guid ControllerRegistryId { get; set; }
        public ControllerRegistry ControllerRegistry { get; set; }
        public string ImageUrl { get; set; }
        public string BlobName { get; set; }
        public long CounterValue { get; set; }
        public bool ProcessedSuccessful { get; set; }

        [MaxLength(256)]
        public string CreatedBy { get; set; }
        [MaxLength(256)]
        public string UpdatedBy { get; set; }
        public DateTime UpdatedDate { get; set; }
        public DateTime CreatedDate { get; set; }
    }
}

File: src/MonitoringSpaApp/MonitoringSpa/ClientApp/src/app/models/telemetry-record.ts
export class TelemetryRecord {

    constructor(id: string, controllerRegistryId?: string, controllerRegistry?: any, imageUrl?: string, counterValue?: number, processedSuccessful?: boolean, createdDate?: string) {
        this.id = id;
        this.controllerRegistryId = controllerRegistryId;
        this.controllerRegistry = controllerRegistry;
        this.imageUrl = imageUrl;
        this.counterValue = counterValue;
        this.processedSuccessful = processedSuccessful;
        this.createdDate = createdDate;
    }

    id: string;
    controllerRegistryId: string;
    controllerRegistry: any;
    imageUrl: string;
    counterValue: number;
    processedSuccessful: boolean;
    createdDate: string;
}
